nixos/modules/services/web-servers/nginx/vhost-options.nix

```diff
@@ -70,12 +66,24 @@ with lib;
       '';
     };
 
-    enableSSL = mkOption {
+    addSSL = mkOption {
       type = types.bool;
       default = false;
       description = "Whether to enable SSL (https) support.";
     };
 
+    onlySSL = mkOption {
+      type = types.bool;
+      default = false;
+      description = "Whether to enable SSL (https) support and reject plain (http) connections.";
```

Comment:
Description proposal:

    Whether to add a separate nginx server block that permanently redirects (301)
    all plain HTTP traffic to HTTPS. This will set defaults for <literal>listen</literal>
    to listen on all interfaces on port 443.